### ISSUE TITLE
Organizer not correctly filtered when dates and topics are used

### ISSUE THREAD

**Issue by FLeudts** [oliverklee/ext-seminars#219]
As far as I know I haven't changed how seminar-topics are created so they don't get an organizer attached but the dates get that option. I don't know if that is the intention but as a result if I filter by organizer no date is displayed as this code in the \BagBuilder\Event.php assumes that the topics have an organizer attached. 

`        $this->whereClauseParts['eventsWithOrganizers'] =
            '((object_type = ' . \Tx_Seminars_Model_Event::TYPE_COMPLETE . ') ' .
            'AND (tx_seminars_seminars.uid IN (' . $eventUids .
            ')) OR (' .
            '(object_type = ' . \Tx_Seminars_Model_Event::TYPE_DATE . ') AND (' .
            'tx_seminars_seminars.topic IN (' . $eventUids . ')))' .
            ')';
`

**Comment by oliverklee** [comment]
I don't follow you. If I read the code correctly, it is about the events being either complete (stand-alone) events or dates, not topics. Am I missing something here?

**Comment by FLeudts** [comment]
yes if the event is a date then it searches for a topic that has this organizer associated with it. 
And in the TCA the topics don't have the organizer as possible entry field. 
So any date that is supposed to be filtered by organizer won't be shown. 
Either the TCA needs to change or the comparison in mysql needs to compare against the uid in both cases (and not against the topic in the second part).

**Comment by oliverklee** [comment]
Then we need the latter (as topics cannot and should not have an organizer). @FLeudts Would you be willing to create a pull request with a fix and a regression test?

**Comment by FLeudts** [comment]
I would be willing to create the pull request. BUt have to take a closer look at the testing side first.

**Comment by oliverklee** [comment]
If you need help with the tests, I'm willing to help (with explanations, links etc.).

**Comment by FLeudts** [comment]
Yes I looked at the tests for a bit yesterday and saw that there is quite a difference between the newer stuff that uses nimut/testing-framework and the older one. Which should I use? Should I convert the limitToOrganizer Tests?

**Comment by oliverklee** [comment]
The least work would be if you add additional tests to the existing `Tests/LegacyUnit/BagBuilder/EventTest.php`. It would be more work if your converted the `limitToOrganizer` tests first, which would be the cleaner approach. So the first approach would be okay, and the second would be great. :-)

**Comment by oliverklee** [comment]
And my plan is to convert all existing test to the nimut TF bit by bit so I can remove the dependency on the phpunit extension at some time.
